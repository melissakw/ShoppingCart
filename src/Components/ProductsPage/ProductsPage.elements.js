import styled from "styled-components";

export const ProductCardContainer = styled.div`
  padding: 10px;
`;
