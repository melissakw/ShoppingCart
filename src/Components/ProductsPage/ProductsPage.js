import React from "react";
import Product from "../Product/Product";

const ProductsPage = ({ products }) => {
  return (
    <div>
      <Product products={products} />
    </div>
  );
};

export default ProductsPage;
