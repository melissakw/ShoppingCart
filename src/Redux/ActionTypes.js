export const ADD_ITEM = "ADD_ITEM";
export const REMOVE_ITEM = "REMOVE_ITEM";
export const EDIT_ITEM_QTY = "EDIT_ITEM_QTY";

export const ADD_TO_CART_LOADING = "ADD_TO_CART_LOADING";
export const ADD_TO_CART_FAILED = "ADD_TO_CART_FAILED";
