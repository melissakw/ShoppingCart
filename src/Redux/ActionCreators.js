import * as ActionTypes from "./ActionTypes";

export const postToCart = (product) => (dispatch) => {
  const productWithQty = {
    id: product.id,
    title: product.title,
    price: product.price,
    image: product.image,
    quantity: 1, //defualt quantity to 1 when adding item to cart
    stock: 10
  };
  dispatch(addToCart(productWithQty));
};

export const addToCart = (product) => ({
  type: ActionTypes.ADD_ITEM,
  payload: product
});

export const removeItem = (id) => ({
  type: ActionTypes.REMOVE_ITEM,
  payload: id
});

export const editItem = (product, editedQuantity) => ({
  type: ActionTypes.EDIT_ITEM_QTY,
  product: product,
  quantity: editedQuantity
});

export const addToCartLoading = () => ({
  type: ActionTypes.ADD_TO_CART_LOADING
});

export const addToCartFailed = (errmess) => ({
  type: ActionTypes.ADD_TO_CART_FAILED,
  payload: errmess
});
