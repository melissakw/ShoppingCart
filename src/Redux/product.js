import * as ActionTypes from "./ActionTypes";

export const product = (
  state = {
    isLoading: true,
    errMess: null,
    cartItems: []
  },
  action
) => {
  switch (action.type) {
    case ActionTypes.ADD_ITEM:
      let duplicateItem = false;
      let copyItems = state.cartItems.map((item) => {
        if (item.id === action.payload.id) {
          item.quantity = Math.min(item.quantity + 1, item.stock);
          duplicateItem = true;
        }
        return item;
      });

      if (duplicateItem) {
        return { ...state, cartItems: copyItems };
      } else {
        return {
          ...state,
          cartItems: [action.payload].concat(state.cartItems)
        };
      }

    case ActionTypes.REMOVE_ITEM:
      return {
        ...state,
        cartItems: state.cartItems.filter((item) => item.id !== action.payload)
      };
    case ActionTypes.EDIT_ITEM_QTY:
      let updatedProductWithNewQty = state.cartItems.map((item) => {
        if (item.id === action.product.id) {
          item.quantity = action.quantity;
        }
        return item;
      });
      return { ...state, cartItems: updatedProductWithNewQty };

    default:
      return state;
  }
};
