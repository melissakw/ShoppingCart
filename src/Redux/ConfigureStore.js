import { createStore, applyMiddleware } from "redux";
import { persistStore, persistCombineReducers } from "redux-persist";
import storage from "redux-persist/lib/storage";
import thunk from "redux-thunk";
import logger from "redux-logger";

import { product } from "./product";

export const ConfigureStore = () => {
  const config = {
    key: "root",
    storage: storage,
    debug: true
  };

  const store = createStore(
    persistCombineReducers(config, {
      product
    }),
    applyMiddleware(thunk, logger)
  );

  const persistor = persistStore(store);
  return { persistor, store };
};
