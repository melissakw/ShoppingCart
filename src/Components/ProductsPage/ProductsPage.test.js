import React from 'react';
import ProductsPage from './ProductsPage';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from "react-redux";
import { ConfigureStore } from "../../Redux/ConfigureStore";
import { BrowserRouter } from 'react-router-dom';

jest.mock('../Product/Product', () => () => 'Product');

test("renders ProductsPage correctly", () => {
	const { store } = ConfigureStore();
  const component = renderer.create(<ProductsPage />);
	const tree = renderer.create(
		<Provider store={store}>
			<BrowserRouter>
				<ProductsPage />
			</BrowserRouter>
		</Provider>
	).toJSON();
	expect(tree).toMatchSnapshot();

	expect(component.root.findByType(ProductsPage).props).toEqual({});
});
