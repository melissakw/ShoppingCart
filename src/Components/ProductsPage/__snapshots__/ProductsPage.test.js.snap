// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductsPage correctly 1`] = `
<div>
  Product
</div>
`;
